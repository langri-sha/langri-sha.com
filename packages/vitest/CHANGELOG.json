{
  "name": "@langri-sha/vitest",
  "entries": [
    {
      "date": "Sat, 27 Jul 2024 15:50:17 GMT",
      "version": "0.1.1",
      "tag": "@langri-sha/vitest_v0.1.1",
      "comments": {
        "none": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "d542415a6b2c2c339a8367c77b925842000f9f9a",
            "comment": "feat(workspace): Exclude compiling test modules"
          }
        ]
      }
    },
    {
      "date": "Fri, 26 Jul 2024 21:00:29 GMT",
      "version": "0.1.1",
      "tag": "@langri-sha/vitest_v0.1.1",
      "comments": {
        "none": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "d58e2b444358fb65166d0878c185220c6e0a7bb0",
            "comment": "feat(workspace): Migrate documentation pathnames"
          },
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "d58e2b444358fb65166d0878c185220c6e0a7bb0",
            "comment": "feat(projen): Add Git ignore patterns"
          }
        ]
      }
    },
    {
      "date": "Wed, 17 Jul 2024 10:48:15 GMT",
      "version": "0.1.1",
      "tag": "@langri-sha/vitest_v0.1.1",
      "comments": {
        "patch": [
          {
            "author": "email not defined",
            "package": "@langri-sha/vitest",
            "commit": "c4b7573f367c165d7c67ed5f259ad53ec9ea5672",
            "comment": "chore(deps): chore(deps): update dependency vitest to v2"
          },
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "c4b7573f367c165d7c67ed5f259ad53ec9ea5672",
            "comment": "fix(workspace): Use correct GitHub directory"
          }
        ],
        "none": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "c4b7573f367c165d7c67ed5f259ad53ec9ea5672",
            "comment": "feat(root): Migrate to ESM"
          }
        ]
      }
    },
    {
      "date": "Mon, 08 Jul 2024 21:43:05 GMT",
      "version": "0.1.0",
      "tag": "@langri-sha/vitest_v0.1.0",
      "comments": {
        "minor": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "feat(workspace): Use `vitest`"
          },
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "feat(vitest): Add `nock`"
          }
        ],
        "none": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "feat(workspace): Add repository path"
          },
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "feat(tsconfig): Use `${configDir}`"
          }
        ],
        "patch": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "fix(projen): Use correct output directory"
          },
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "fix(projen): Use workspace `@langri-sha/tsconfig`"
          }
        ]
      }
    }
  ]
}
