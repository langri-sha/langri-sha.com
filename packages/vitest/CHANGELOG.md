# Change Log - @langri-sha/vitest

This log was last generated on Wed, 17 Jul 2024 10:48:15 GMT and should not be manually modified.

<!-- Start content -->

## 0.1.1

Wed, 17 Jul 2024 10:48:15 GMT

### Patches

- chore(deps): chore(deps): update dependency vitest to v2 (email not defined)
- fix(workspace): Use correct GitHub directory (filip.dupanovic@gmail.com)

## 0.1.0

Mon, 08 Jul 2024 21:43:05 GMT

### Minor changes

- feat(workspace): Use `vitest` (filip.dupanovic@gmail.com)
- feat(vitest): Add `nock` (filip.dupanovic@gmail.com)

### Patches

- fix(projen): Use correct output directory (filip.dupanovic@gmail.com)
- fix(projen): Use workspace `@langri-sha/tsconfig` (filip.dupanovic@gmail.com)
