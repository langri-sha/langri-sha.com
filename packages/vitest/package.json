{
  "name": "@langri-sha/vitest",
  "version": "0.0.0",
  "bugs": {
    "url": "https://github.com/langri-sha/langri-sha.com/issues"
  },
  "repository": {
    "url": "langri-sha/langri-sha.com",
    "path": "/packages/vitest"
  },
  "license": "MIT",
  "author": {
    "name": "Filip DupanoviÄ‡",
    "email": "filip.dupanovic@gmail.com",
    "url": "https://langri-sha.com",
    "organization": false
  },
  "type": "module",
  "main": "src/index.ts",
  "scripts": {
    "prepublishOnly": "rm -rf dist; tsc --project tsconfig.build.json"
  },
  "dependencies": {
    "nock": "13.5.4",
    "tempy": "1.0.1"
  },
  "devDependencies": {
    "@langri-sha/tsconfig": "workspace:*"
  },
  "peerDependencies": {
    "vitest": "^1.6.0"
  },
  "publishConfig": {
    "access": "public",
    "main": "dist/index.js",
    "types": "dist/index.d.ts"
  },
  "//": "~~ Generated by projen. To modify, edit .projenrc.mts and run \"npx projen\"."
}
