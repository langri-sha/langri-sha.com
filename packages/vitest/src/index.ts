export * from 'vitest'
export { default as nock } from 'nock'
export { default as tempy } from 'tempy'
