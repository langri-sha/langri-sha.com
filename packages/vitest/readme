# @langri-sha/vitest

Provides some useful helpers that are commonly used for authoring tests.

## Usage

Install the required dependencies:

```sh
npm install -D vitest @langri-sha/vitest
```

Then import your Vitest dependencies from here:

```js
// some.test.js
import { expect, test, tempy } from '@langri-sha/vitest'

test(/*...*/)
```

## See

- [`@jest/globals`]
- [`tempy`]

[`@jest/globals`]: https://jestjs.io/docs/api
[`tempy`]: https://github.com/sindresorhus/tempy
