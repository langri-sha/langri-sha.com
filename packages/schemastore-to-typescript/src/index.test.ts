import * as path from 'node:path'
import * as fs from 'node:fs/promises'
import { afterEach, expect, nock, tempy, test } from '@langri-sha/vitest'
import { compile } from './index.js'
import { vi } from 'vitest'
import envPaths from 'env-paths'

vi.mock('env-paths', () => {
  const cache = tempy.directory()

  return {
    default: () => ({ cache }),
  }
})

const paths = envPaths('schemastore-to-typescript')

afterEach(() => {
  nock.cleanAll()
})

test('compiles to TypeScript', async () => {
  nock('https://json.schemastore.org')
    .get('/some')
    .reply(
      200,
      {
        $schema: 'http://json-schema.org/draft-07/schema#',
        title: 'Some Schema',
        type: 'object',
        properties: {
          id: {
            type: 'string',
            description: 'The unique identifier for some item.',
          },
        },
        required: ['id'],
      },
      {
        'Cache-Control': 'public, no-store',
      },
    )

  const ts = await compile('some')

  expect(ts).toMatchInlineSnapshot(`
    "/* eslint-disable */
    /**
     * This file was automatically generated by json-schema-to-typescript.
     * DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
     * and run json-schema-to-typescript to regenerate this file.
     */

    export interface SomeSchema {
      /**
       * The unique identifier for some item.
       */
      id: string;
      [k: string]: unknown;
    }
    "
  `)
})

test('uses cache', async () => {
  nock('https://json.schemastore.org')
    .get('/foobar')
    .reply(
      200,
      {
        $schema: 'http://json-schema.org/draft-07/schema#',
        title: 'Foobar Schema',
        type: 'object',
        properties: {
          id: {
            type: 'string',
            description: 'The unique identifier for a foobar item.',
          },
        },
        required: ['id'],
      },
      {
        'Cache-Control': 'public, max-age=3600',
      },
    )
    .get('/foobar')
    .reply(
      200,
      {
        $schema: 'http://json-schema.org/draft-07/schema#',
        title: 'Barbaz Schema',
        type: 'object',
        properties: {
          id: {
            type: 'string',
            description: 'The unique identifier for a barbaz item.',
          },
        },
        required: ['id'],
      },

      {
        'Cache-Control': 'public, max-age=3600',
        ETag: 'etag123',
      },
    )

  expect(await compile('foobar')).toBe(await compile('foobar'))

  await new Promise((resolve) => {
    setTimeout(resolve, 0)
  })

  const cached = JSON.parse(
    (await fs.readFile(path.join(paths.cache, 'requests.json'))).toString(
      'utf8',
    ),
  )

  expect(cached.cache).toHaveLength(1)
  expect(cached.cache[0][0]).toBe(
    'keyv:GET:https://json.schemastore.org/foobar',
  )
})

test('handles missing schemas', async () => {
  nock('https://json.schemastore.org').get('/unknown').reply(404)

  expect(compile('unknown')).rejects.toThrow(/Couldn't find schema unknown/)
})

test('handles missing schemas', async () => {
  nock('https://json.schemastore.org')
    .get('/err')
    .replyWithError('Something went wrong')

  expect(compile('err')).rejects.toThrow(/Something went wrong/)
})
