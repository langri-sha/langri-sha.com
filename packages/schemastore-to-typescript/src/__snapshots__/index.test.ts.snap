// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`compiles to TypeScript 1`] = `
"/* eslint-disable */
/**
 * This file was automatically generated by json-schema-to-typescript.
 * DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
 * and run json-schema-to-typescript to regenerate this file.
 */

export interface FoobarSchema {
  /**
   * The unique identifier for a foobar item.
   */
  id: string;
  [k: string]: unknown;
}
"
`;
